### PR TITLE
Fix typos in module/zfs/

### PR DESCRIPTION
**Pull request by Gelma**

### CODE REVIEW

File: module/zfs/metaslab.c [hunk 1]
```diff
@@ -4433,7 +4433,7 @@ metaslab_group_alloc_normal(metaslab_group_t *mg, zio_alloc_list_t *zal,
 			/*
 			 * Even though we don't hold the ms_lock for the
 			 * primary metaslab, those fields should not
-			 * change while we hold the mg_lock. Thus is is
+			 * change while we hold the mg_lock. Thus is
```
[the remark this answers is not in the document]
Comment (Gelma) [reply module/zfs/metaslab.c]: Done!

**Comment by Gelma** [comment]
> Thanks for doing this! I was surprised by the amount of typos that have crept in over the years. Out of curiosity, how did you find all of these? Did you use a spelling/grammar checker on the comments?

I have stupid/messy Python and Bash scripts I wrote in the last five years. They run machine learning, deep mind, IA... no, no, they are made of stupid mix of regex,  black/white list, some tricks (if a word appears just one time, probably it's a typo, or a not used variable/include, and so on).

I take note of already seen typos, because we usually repeat our mistakes, parsing git logs of projects on github (another stupid script), or from what I already fixed.

All of these (via PostgreSQL and so on) generates a few output, very easy to read and check (for me), with colors and so on.

I usually give back to community my dict (look at Codespell for example; I send them all my fixed typos of LibreOffice and project I follow/use).
The code is really messy and keep changing. At the moment having it up and running on something else than my laptop is a nightmare. When it will be really user friendly, I'm going to GPL'ed.

File: module/zfs/zio.c [hunk 2]
```diff
@@ -4568,7 +4568,7 @@ zio_done(zio_t *zio)
 		/*
 		 * "The Godfather" I/O monitors its children but is
 		 * not a true parent to them. It will track them through
-		 * the pipeline but severs its ties whenever they get into
+		 * the pipeline but serves its ties whenever they get into
```
[the remark this answers is not in the document]
Comment (Gelma) [reply module/zfs/zio.c]: Reverted and squashed!

File: module/zfs/zio_crypt.c [hunk 3]
```diff
@@ -369,7 +369,7 @@ zio_crypt_key_get_salt(zio_crypt_key_t *key, uint8_t *salt)
 /*
  * This function handles all encryption and decryption in zfs. When
  * encrypting it expects puio to reference the plaintext and cuio to
- * reference the cphertext. cuio must have enough space for the
+ * reference the cyphertext. cuio must have enough space for the
```
Comment (rlaager) [inline module/zfs/zio_crypt.c]: The correct (en_US) spelling is ciphertext.

Comment (Gelma) [reply module/zfs/zio_crypt.c]: Thanks! For Wikipedia they are both valid:
«ciphertext or cyphertext is the result of encryption performed on plaintext»

You tell me if we can leave it alone or change it.

Comment (Gelma) [reply module/zfs/zio_crypt.c]: "Thanks" was for the review. Not ironic.

Comment (rlaager) [reply module/zfs/zio_crypt.c]: I grepped. There are no other instances of "cypher". Please use "ciphertext".

Comment (Gelma) [reply module/zfs/zio_crypt.c]: Done!

**Pull request merged**